src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -96,17 +109,27 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             var tagHelpersFromCompilation = context.CompilationProvider
                 .Combine(generatedDeclarationSyntaxTrees.Collect())
                 .Combine(razorSourceGeneratorOptions)
+                .WithLambdaComparer(static (a, b) =>
```

Comment:
Consider extracting this logic out into either an `IValueComparer<Tuple<object, RazorSourceGenerationOptions>>` or another extension method, so you don't have to duplicate the logic.